### ISSUE TITLE
driver: bad connection and other errors on SQL Server 2012

### ISSUE THREAD

**Issue by clns** [denisenkom/go-mssqldb#105]
I get `sql: no rows in result set` even if the query is completely invalid.

So I decided to run the tests to see what the output is. I created an empty `test` database and I'm using windows authentication. The SQL Server `@@VERSION` prints:

```
Microsoft SQL Server 2012 (SP1) - 11.0.3153.0 (X64) 
    Jul 22 2014 15:26:36 
    Copyright (c) Microsoft Corporation
    Express Edition (64-bit) on Windows NT 6.2 <X64> (Build 9200: )
```

The tests fail with errors like:

```
driver: bad connection
Login error: WSARecv tcp 127.0.0.1:51445: An existing connection was forcibly closed by the remote host.
Commit failed WSARecv tcp 127.0.0.1:52057: An established connection was aborted by the software in your host machine.
```

You can see the output of the `go test` command in this gist: [full output](https://gist.github.com/clns/98766b283fb731e12c79#file-go-test-all-ps1), [stdout-only](https://gist.github.com/clns/98766b283fb731e12c79#file-go-test-stdout-ps1), [stderr-only](https://gist.github.com/clns/98766b283fb731e12c79#file-go-test-stderr-ps1).

I also noticed that if I use `;log=8` (or greater than 8, when there's some logging, smaller values don't log anything for a SELECT statement) in the connection string, I always get _driver: bad connection_ error.

**Comment by clns** [comment]
Apparently these issues happen when using IPv6. On Windows, [`localhost` points to `::1`](http://superuser.com/questions/436574/ipv4-vs-ipv6-priority-in-windows-7/436944#436944), so if I change from `localhost` to `127.0.0.1` in the connection string it works fine.

Maybe this issue then is about supporting IPv6. I'd be happy to hear some input on this.

**Comment by blitzd** [comment]
It seems like using any sort of short or FQDN results in an IPv6 connection being used rather than IPv4.

**Comment by zbliang** [comment]
password:sa
 port:1433
 server:192.168.0.128
 user:sa
 connString:server=192.168.0.128;user id=sa;password=sa;port=1433
2015/10/20 14:24:19 Prepare failed:Login error: WSARecv tcp 192.168.2.20:53117: An existing connection was forcibly closed by the remote host.

my ip is 192.168.2.20

**Comment by brydavis** [comment]
I'm experience same error `driver: bad connection`.

Issue seems to arise when query data from large table (5+ million rows).

**Comment by melignus** [comment]
@brydavis I ran into this issue when running a report. It wasn't a million rows, but it's a huge number of columns and a slow running report. What fixed it for me was just increasing the "connection timeout=" parameter in the connection string. I noticed that it defaults to 30 seconds in via the README and increasing it fixed my problem.

I agree that `driver: bad connection` wasn't a very telling error. `driver: connection timeout` might be more appropriate? I see the timeout being passed to the Dial calls but I don't see it being checked anywhere, I'll take a dive and see if I can come up with something.
